### ISSUE TITLE
using the new Rock 5 Model B

### ISSUE THREAD

**Issue by slaesh** [balena-os/balena-radxa#221]
is this possible?

**Comment by floion** [comment]
At this time, no. Even the vendor has not released yocto support for it afaics: https://github.com/radxa/meta-radxa/tree/dunfell/conf/machine (there's no conf file for the Rock 5 there)

**Comment by RadxaYuntian** [comment]
We have switched to meta-rockchip for new development: https://github.com/radxa/meta-rockchip/blob/kirkstone-radxa/conf/machine/rockchip-rk3588-rock-5b.conf

**Comment by slaesh** [comment]
Thanks @RadxaYuntian .. does this help @floion ? =)

**Comment by terehov** [comment]
I think this is the perfect hardware for many use cases. We are quite unhappy with the performance of Raspberry 4 and version 5 seems far away. 
I believe this would be a huge benefit for the balena system.

**Comment by floion** [comment]
@slaesh @terehov do any of you have the hw in hand? On our side we do not have it at this moment.
If you do have the hardware, would you be interested in adding support for it? We have instructions on what needs to be done https://github.com/balena-os/meta-balena/blob/master/contributing-device-support.md

**Comment by slaesh** [comment]
yes I have. but what is missing is.. `This will require knowledge of the Yocto project and familiarity with tools used to build custom images` :sweat: 

but I will give it a shoot if I have some spare-time soon

**Comment by floion** [comment]
Hi @RadxaYuntian which is the official kirkstone branch that should be used on the new layer? There are a few there: `kirkstone`, `kirkstone-radxa`, `kirkstone-radxa-rebased`.
Thank you

**Comment by RadxaYuntian** [comment]
For future Yocto related question you can contact @RadxaStephen directly, since he is the current Yocto maintainer in Radxa.

The official branch is `kirkstone-radxa`.

**Comment by csvke** [comment]
> For future Yocto related question you can contact @RadxaStephen directly, since he is the current Yocto maintainer in Radxa.
> 
> The official branch is `kirkstone-radxa`.

@RadxaStephen I have started to create Balena support for rock5b and here's the progress
https://github.com/frankieslab/balena-radxa/commit/2eff98e26a71c613ced01e1f96e1b01eb9660948

However, I wonder if there's any reference I can follow when I prepare `recipes-core/images` https://docs.balena.io/reference/OS/customer-board-support/#recipes-coreimages-directory

Should I start base on rock pi 4b?

**Comment by floion** [comment]
> > For future Yocto related question you can contact @RadxaStephen directly, since he is the current Yocto maintainer in Radxa.
> > The official branch is `kirkstone-radxa`.
> 
> @RadxaStephen I have started to create Balena support for rock5b and here's the progress [frankieslab@2eff98e](https://github.com/frankieslab/balena-radxa/commit/2eff98e26a71c613ced01e1f96e1b01eb9660948)
> 
> However, I wonder if there's any reference I can follow when I prepare `recipes-core/images` https://docs.balena.io/reference/OS/customer-board-support/#recipes-coreimages-directory
> 
> Should I start base on rock pi 4b?

Yes, you can look at the Rock Pi 4 B as a reference

**Comment by floion** [comment]
Hi @RadxaYuntian I can't find the older machines in the new layer here: https://github.com/radxa/meta-rockchip/tree/kirkstone-radxa/conf/machine
So the RockPi 4B will not have kirkstone support?

**Comment by RadxaYuntian** [comment]
You will have to use 4SE on `kirkstone`, but there is one for 4B in [`scarthgap`](https://github.com/radxa/meta-rockchip/blob/scarthgap/conf/machine/rock-pi-4b.conf).

**Comment by floion** [comment]
@RadxaYuntian so the conf machine for the 4SE on the `kirkstone` branch can be used to generate a working image for the RockPi 4B?

**Comment by RadxaYuntian** [comment]
Those 2 models are software compatible.

**Comment by floion** [comment]
@RadxaYuntian ok so going forward with `scarthgap` and newer, there will not be a machine for the 4SE but rather only for the 4B, right?

**Comment by RadxaYuntian** [comment]
You need to ask @RadxaStephen since he is the maintainer of Yocto layer.

**Comment by akshar001** [comment]
@slaesh check @terehov [Vicharak](https://vicharak.in). We've added support of BalenaOS for our RK3588 based board.

https://discuss.vicharak.in/t/balenaos-support-enabled-for-rk3588-platform-flash-and-deploy-docker-application-guide-on-vicharaks-axon-single-board-computer/195
